etc/eslint/rules/typescript.js

```diff
@@ -0,0 +1,2526 @@
+
+/**
+* Requires `for-in` loops to be filtered with an `if` statement.
+*
+* @name guard-for-in
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [guard-for-in]{@link https://eslint.org/docs/rules/guard-for-in}
+*
+* @example
+* // Bad...
+* for (const key in object) {
+*   doSomething(object[key]);
+* }
+*
+* // Good...
+* for (const key in object) {
+*   if (Object.prototype.hasOwnProperty.call(object, key)) {
+*     doSomething(object[key]);
+*   }
+* }
+*/
+rules[ 'guard-for-in' ] = 'error';
+
+/**
+* Requires identifiers to match a specified regular expression.
+*
+* @name id-match
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [id-match]{@link https://eslint.org/docs/rules/id-match}
+*
+* @example
+* // Bad...
+* const my_var = 'bad';
+*
+* // Good...
+* const myVar = 'good';
+*/
+rules[ 'id-match' ] = 'error';
+
+/**
+* Disallows default exports in modules.
+*
+* @name import/no-default-export
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [import/no-default-export]{@link https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-default-export.md}
+*
+* @example
+* // Okay...
+* export default function myFunction() { }
+*/
+rules[ 'import/no-default-export' ] = 'off';
+
+/**
+* Reports use of a deprecated name.
+*
+* @name import/no-deprecated
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [import/no-deprecated]{@link https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-deprecated.md}
+*
+* @example
+* // Bad...
+* import { oldFunction } from 'lib'; // if oldFunction is deprecated
+*
+* // Good...
+* import { newFunction } from 'lib';
+*/
+rules[ 'import/no-deprecated' ] = 'error';
+
+/**
+* Ensures that there are no extraneous dependencies being imported.
+*
+* @name import/no-extraneous-dependencies
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [import/no-extraneous-dependencies]{@link https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-extraneous-dependencies.md}
+*
+* @example
+* // Okay...
+* import 'some-optional-module';
+*/
+rules[ 'import/no-extraneous-dependencies' ] = 'off';
+
+/**
+* Prevents importing the submodules of other modules.
+*
+* @name import/no-internal-modules
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [import/no-internal-modules]{@link https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-internal-modules.md}
+*
+* @example
+* // Okay...
+* import something from 'module/some/internal/path';
+*/
+rules[ 'import/no-internal-modules' ] = 'off';
+
+/**
+* Forbids import statements that import nothing.
+*
+* @name import/no-unassigned-import
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [import/no-unassigned-import]{@link https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-unassigned-import.md}
+*
+* @example
+* // Bad...
+* import 'myModule';
+*
+* // Good...
+* import 'myPolyfill';
```

Comment:
I am not sure how this is different than the "bad" example here.

Reply:
Changed the example.